web/modules/custom/joinup_core/joinup_core.post_update.php

```diff
@@ -6,3 +6,55 @@
  */
 
 declare(strict_types = 1);
+
+use Drupal\solution\Entity\SolutionInterface;
+
+/**
+ * Remove a selected set of solutions and 1 document from Eurostat.
+ */
+function joinup_core_post_update_0106300(): void {
+  /** @var \Drupal\sparql_entity_storage\SparqlEntityStorageInterface $rdf_storage */
+  $rdf_storage = \Drupal::entityTypeManager()->getStorage('rdf_entity');
+  /** @var \Drupal\collection\Entity\CollectionInterface $collection */
+  $collection = $rdf_storage->load('http://data.europa.eu/w21/01b52000-c0ba-4e64-a32c-79557a743462');
+  $solutions = array_filter($collection->getSolutions(), function (SolutionInterface $solution) {
+    // Remove all solutions starting with the word SCL.
+    if (substr($solution->label(), 0, 4) === 'SCL ') {
+      return TRUE;
+    }
+
+    // Additionally remove 3 specific solutions.
+    return in_array($solution->id(), [
+      // 'General information on the "Standard code lists" project'.
+      'http://ec.europa.eu/eurostat/ramon/miscellaneous/index.cfm?TargetUrl=DSP_GENINFO_SCL',
+      // 'Eurostat SDMX Converter'.
+      'http://data.europa.eu/w21/35ef35c6-a530-4821-8c52-8d796ec86a1d',
+      // 'Eurostat XBRL Reference Architecture'.
+      'http://data.europa.eu/w21/96f74e35-abda-4475-8b2d-22fed2120fc5',
+    ]);
+  });
+
+  $total = count($solutions);
+  $count = 0;
+
+  foreach ($solutions as $solution) {
+    \Drupal::logger('joinup')->notice('[%count/%total] Deleting solution %title', [
+      '%count' => ++$count,
+      '%total' => $total,
+      '%title' => $solution->label(),
+    ]);
+    $solution->delete();
```

Comment:
Also, the script is very slow. I wonder if we cannot do a direct SPARQL query. Or maybe deleting all in a single command would make it faster `$rdf_storage->delete($solutions)`?

Reply:
It is slow because it cleans up all group content, memberships and OG roles alongside with the groups themselves. I have to look into why distributions are not cleaned up. Maybe they are still being returned by Search API?

Reply:
I just checked this in depth, and all group content seems to be successfully cleaned up. With the debugger I can see them being removed in `OgDeleteOrphansBase::deleteOrphan()`. This included all referenced entity types:

- RDF entities
- Nodes
- OG memberships
- OG menus

This is happening in the shutdown hook at the end of the process. Is it perhaps possible your script didn't invoke the shutdown hook?

This is also why I went with the Drupal API rather than direct queries, it becomes quite complicated to clean up all these references and cache tags if we are going to code this by hand.

Regarding the speed of the update, I agree it is slow, but it seems to be within acceptable limits. On my system it takes 4 minutes 42 seconds. I cannot predict the situation on production though.

```
 [success] Finished performing updates.

real	4m42.223s
user	0m24.881s
sys	0m2.975s
```